lib/runner/extensions/request.command.js

```diff
@@ -56,49 +84,46 @@ module.exports = {
-            _.set(payload._item, '__parent', payload.item.__parent);
+                self = this,
+                replayed,
+                context;
+
+            /**
+             * @type {Object}
+             * @property {Item} item -  Holds a copy of the item given in the payload, so that it can be manipulated
+             * as necessary
+             * @property {RequestAuthBase|undefined} auth - If present, is the instance of Auth in the collection, which
+             * is changed as necessary using intermediate requests, etc.
+             * @property {VariableScope} environment
+             * @property {VariableScope} globals
+             * @property {Object} data
+             * @property {Boolean} replay - Indicates whether this request a replayed request
+             */
+            context = createContext(payload);
+            replayed = context.replay;
 
             // Run the helper functions
-            async.applyEachSeries(helpers, payload, self, function (err) {
+            async.applyEachSeries(prehelpers, context, self, function (err) {
                 var xhr,
                     aborted,
-                    item = payload._item,
-
-                    /**
-                     * Helper function which calls the beforeRequest trigger.
-                     *
-                     * @param err
-                     */
-                    beforeRequest = function (err) {
-                        // Trigger the beforeRequest callback
-                        // todo: ensure that the correct item and other params are bubbled up
-                        self.triggers.beforeRequest(err, payload.coords, item.request, payload.item, {
-                            abort: function () {
-                                !aborted && xhr && xhr.abort();
-                                aborted = true;
-                            }
-                        });
-                    },
-
-                    /**
-                     * Helper function to call the afterRequest trigger.
-                     *
-                     * @param err
-                     * @param response
-                     * @param request
-                     * @param cookies
-                     */
-                    afterRequest = function (err, response, request, cookies) {
-                        self.triggers.request(err, payload.coords, response, request, payload.item, cookies);
-                    };
+                    item = context.item,
+                    beforeRequest,
+                    afterRequest;
+
+                // Helper function which calls the beforeRequest trigger ()
+                beforeRequest = function (err) {
+                    !replayed && self.triggers.beforeRequest(err, payload.coords, item.request, payload.item, {
+                        abort: function () {
+                            !aborted && xhr && xhr.abort();
+                            aborted = true;
```

Comment:
This should always happen, whether `aborted` is true or false (can seem very non-intuitive, but better safe than sorry)